pkg/vdri/trustbloc/vdri.go

```diff
@@ -121,18 +126,16 @@ func (v *VDRI) Read(did string, opts ...vdriapi.ResolveOpts) (*docdid.Doc, error
 	var doc *docdid.Doc
 
 	for _, e := range endpoints {
-		sideTreeVDRI, err := v.getHTTPVDRI(e.URL)
+		resp, err := v.sidetreeResolve(e.URL, did, opts...)
 		if err != nil {
-			return nil, fmt.Errorf("failed to create new sidetree vdri: %w", err)
+			return nil, err
 		}
 
-		resp, err := sideTreeVDRI.Read(did, opts...)
-		if err != nil {
-			return nil, fmt.Errorf("failed to resolve did: %w", err)
+		if doc == nil {
+			doc = resp
+		} else if !cmp.Equal(resp, doc) {
+			return nil, fmt.Errorf("endpoints responded with different results for did")
```

Comment:
Note: it is possible to have a mismatch if the consortium is in the midst of an update.
(we will need retries and/or query extra organizations).

CC: @bstasyszyn